sktree/stats/forestht.py

```diff
@@ -0,0 +1,736 @@
+        Controls both the randomness of the bootstrapping of the samples used
+        when building trees (if ``bootstrap=True``) and the sampling of the
+        features to consider when looking for the best split at each node
+        (if ``max_features < n_features``).
+        See :term:`Glossary <random_state>` for details.
+
+    verbose : int, default=0
+        Controls the verbosity when fitting and predicting.
+
+    test_size : float, default=0.2
+        Proportion of samples per tree to use for the test set.
+
+    permute_per_tree : bool, default=True
+        Whether to permute the covariate index per tree or per forest.
+
+    sample_dataset_per_tree : bool, default=False
+        Whether to sample the dataset per tree or per forest.
+
+    Attributes
+    ----------
+    estimator_ : BaseForest
+        The estimator used to compute the test statistic.
+
+    n_samples_test_ : int
+        The number of samples used in the final test set.
+
+    indices_train_ : ArrayLike of shape (n_samples_train,)
+        The indices of the samples used in the training set.
+
+    indices_test_ : ArrayLike of shape (n_samples_test,)
+        The indices of the samples used in the testing set.
+
+    samples_ : ArrayLike of shape (n_samples_final,)
+        The indices of the samples used in the final test set that would slice
+        the original ``(X, y)`` input.
+
+    y_true_final_ : ArrayLike of shape (n_samples_final,)
+        The true labels of the samples used in the final test.
+
+    posterior_final_ : ArrayLike of shape (n_samples_final,)
+        The predicted posterior probabilities of the samples used in the final test.
+
+    null_dist_ : ArrayLike of shape (n_repeats,)
+        The null distribution of the test statistic.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+
+    def __init__(
+        self,
+        estimator=None,
+        random_state=None,
+        verbose=0,
+        test_size=0.2,
+        permute_per_tree=True,
+        sample_dataset_per_tree=True,
+    ):
+        super().__init__(
+            estimator=estimator,
+            random_state=random_state,
+            verbose=verbose,
+            test_size=test_size,
+            permute_per_tree=permute_per_tree,
+            sample_dataset_per_tree=sample_dataset_per_tree,
+        )
+
+    def _get_estimator(self):
+        if self.estimator is None:
+            estimator_ = RandomForestClassifier()
+        elif not isinstance(self.estimator, (ForestClassifier, sklearnForestClassifier)):
+            raise RuntimeError(f"Estimator must be a ForestClassifier, got {type(self.estimator)}")
+        else:
+            # self.estimator is an instance of a ForestEstimator
+            estimator_ = self.estimator
+        return estimator_
+
+    def _statistic(
+        self,
+        estimator: ForestClassifier,
+        X: ArrayLike,
+        y: ArrayLike,
+        covariate_index: ArrayLike = None,
+        metric="mse",
```

Comment:
We should set the default metrics to AUC or MI, as current paper doesn't use MSE.

Reply:
Done